build.rs

```diff
@@ -0,0 +1,14 @@
+use std::process::Command;
+
+fn main() {
+    let rustc_output = Command::new("rustc")
+        .arg("--version")
+        .output()
+        .expect("Failed to execute rustc");
+
+    let rustc_version = String::from_utf8_lossy(&rustc_output.stdout);
+
+    if rustc_version.contains("nightly") {
+        println!("cargo:rustc-cfg=nightly");
+    }
+}
```

Comment:
Let's not do this, the complexity isn't worth it.

Nightly detection is often finicky and causes issues in downstream crates (remember, `cargo-semver-checks` is a library too — that's how `release-plz` integrates it) and is generally not recommended.

This is taking our test suite's inability to specify different expected results for different Rust versions, and pushing the complexity onto our users. It's not a good idea.

Reply:
I see! Okay sounds good 👍